include/af/autograd/Functions.hpp

```diff
@@ -44,6 +44,8 @@ namespace af {
         Variable operator <=(const Variable &lhs, const double &rhs);
 
         Variable operator !(const Variable &input);
+        Variable select_index(const Variable &input, const Variable &idx);
+        Variable set_index(const Variable &input, const Variable &idx, const Variable &vals);
```

Comment:
I think `lookup` and `assign` are better. Or  go with`gather` vs `scatter` like tensorflow.